src/ServiceMonitor/IISConfigUtil.cpp

```diff
@@ -121,10 +121,9 @@ HRESULT IISConfigUtil::Initialize()
     return hr;
 }
 
-HRESULT IISConfigUtil::BuildAppCmdCommand(unordered_map<wstring, wstring> envSet, WCHAR* pstrAppPoolName, wstring** pStrCmd, BOOL fAddCommand)
+HRESULT IISConfigUtil::BuildAppCmdCommand(vector<pair<wstring, wstring>> vecSet, WCHAR* pstrAppPoolName, wstring** pStrCmd, BOOL fAddCommand, int* beginIndex)

```

Comment:
Did you consider a new return value for when more processing is needed because not all the environment variables could be built into the command?